pkg/kptfile/pkgfile_test.go

```diff
@@ -1,926 +0,0 @@
-	"github.com/stretchr/testify/assert"
-	"sigs.k8s.io/kustomize/kyaml/yaml"
-)
-
-// TestReadFile tests the ReadFile function.
-func TestReadFile(t *testing.T) {
-	dir, err := ioutil.TempDir("", fmt.Sprintf("%s-pkgfile-read", testutil.TmpDirPrefix))
-	assert.NoError(t, err)
-	err = ioutil.WriteFile(filepath.Join(dir, KptFileName), []byte(`apiVersion: kpt.dev/v1alpha1
-kind: Kptfile
-metadata:
-  name: cockroachdb
-upstream:
-  type: git
-  git:
-    commit: dd7adeb5492cca4c24169cecee023dbe632e5167
-    directory: staging/cockroachdb
-    ref: refs/heads/owners-update
-    repo: https://github.com/kubernetes/examples
-`), 0600)
-	assert.NoError(t, err)
-
-	f, err := kptfileutil.ReadFile(dir)
-	assert.NoError(t, err)
-	assert.Equal(t, KptFile{
-		ResourceMeta: yaml.ResourceMeta{
-			ObjectMeta: yaml.ObjectMeta{
-				NameMeta: yaml.NameMeta{
-					Name: "cockroachdb",
-				},
-			},
-			TypeMeta: yaml.TypeMeta{
-				APIVersion: TypeMeta.APIVersion,
-				Kind:       TypeMeta.Kind},
-		},
-		Upstream: Upstream{
-			Type: "git",
-			Git: Git{
-				Commit:    "dd7adeb5492cca4c24169cecee023dbe632e5167",
-				Directory: "staging/cockroachdb",
-				Ref:       "refs/heads/owners-update",
-				Repo:      "https://github.com/kubernetes/examples",
-			},
-		},
-	}, f)
-}
-
-// TestReadFile_failRead verifies an error is returned if the file cannot be read
-func TestReadFile_failRead(t *testing.T) {
-	dir, err := ioutil.TempDir("", fmt.Sprintf("%s-pkgfile-read", testutil.TmpDirPrefix))
-	assert.NoError(t, err)
-	err = ioutil.WriteFile(filepath.Join(dir, " KptFileError"), []byte(`apiVersion: kpt.dev/v1alpha1
-kind: Kptfile
-metadata:
-  name: cockroachdb
-upstream:
-  type: git
-  git:
-    commit: dd7adeb5492cca4c24169cecee023dbe632e5167
-    directory: staging/cockroachdb
-    ref: refs/heads/owners-update
-    repo: https://github.com/kubernetes/examples
-`), 0600)
-	assert.NoError(t, err)
-
-	f, err := kptfileutil.ReadFile(dir)
-	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "no such file or directory")
-	assert.Equal(t, KptFile{}, f)
-}
-
-// TestReadFile_failUnmarshal verifies an error is returned if the file contains any unrecognized fields.
-func TestReadFile_failUnmarshal(t *testing.T) {
-	dir, err := ioutil.TempDir("", fmt.Sprintf("%s-pkgfile-read", testutil.TmpDirPrefix))
-	assert.NoError(t, err)
-	err = ioutil.WriteFile(filepath.Join(dir, KptFileName), []byte(`apiVersion: kpt.dev/v1alpha1
-kind: Kptfile
-metadata:
-  name: cockroachdb
-upstreamBadField:
-  type: git
-  git:
-    commit: dd7adeb5492cca4c24169cecee023dbe632e5167
-    directory: staging/cockroachdb
-    ref: refs/heads/owners-update
-    repo: https://github.com/kubernetes/examples
-`), 0600)
-	assert.NoError(t, err)
-
-	f, err := kptfileutil.ReadFile(dir)
-	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "upstreamBadField not found")
-	assert.Equal(t, KptFile{}, f)
-}
-
-func TestKptFile_MergeOpenAPI(t *testing.T) {
```

Comment:
Why are all these tests being deleted? The code is just moved, so I don't see any reason why we would delete the tests.

Reply:
Only TestKptFile_MergeOpenAPI is deleted as they are no longer relevant. Other tests are moved to util_test.go

Reply:
But we do copy over the `MergeOpenAPI` code as far as I can tell (https://github.com/GoogleContainerTools/kpt/pull/1479/files#diff-5a6bded448fa9cf76ddb020aef86f264a39ac465eb22d05b2533e0e48c31bb3cR180). We should either delete both the code and the tests or keep both.
And all the tests for merging subpackages doesn't seem to be in util_test.go.

Reply:
Deleted `MergeOpenAPI` and related methods.